configure.ac

```diff
@@ -1138,6 +1138,9 @@ fi
 if test x$enable_wallet != xno; then
     dnl Check for libdb_cxx only if wallet enabled
     BITCOIN_FIND_BDB48
+
+    dnl Check for sqlite3
+    AC_CHECK_HEADERS([sqlite3.h], [AC_CHECK_LIB([sqlite3], [sqlite3_open], [SQLITE_LIBS=-lsqlite3], [AC_MSG_ERROR(sqlite3_open missing from libsqlite3)], [-pthread -lpthread])], [AC_MSG_ERROR(sqlite3.h headers missing)])
```

Comment:
In commit "Add libsqlite3" (31243f5482bb1c8a71affbe7ced6653a09bd6829)

Would suggest splitting this commit and other build and depends and travis related commits into a separate build PR so it can get feedback from bitcoin build aficionados (and so this PR can more approachable for regular and wallet reviewers).

I think probably build reviewers will want a `--with-sqlite` configure option to allow sqlite to be disabled in the build even if it is present in the system. They might also want the sqlite location to be determined through pkgconfig instead of assumed to be in the system include directory.

Reply:
I attempted to make it so that BDB or sqlite could be disabled, but I wasn't able to get it to work.